File: 0409-longest-palindrome/0409-longest-palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        o = {}
        
        for i in range(len(s)):
            o[s[i]] = o.get(s[i], 0) + 1
        
        numbers = list(o.values())
        evens = list(filter(lambda x: x % 2 == 0, numbers))
        odds = list(filter(lambda x: x % 2 == 1, numbers))
        
        result = 0
        if evens:
            result += sum(evens)
            
        if odds:
            gross = sum(odds)
            maximum = max(odds)
            gross -= maximum
            
            if gross != 0 :
                gross -= len(odds) - 1
            
            result += gross
            result += maximum
            
        return result

File: 0409-longest-palindrome/0409-longest-palindrome.swift
class Solution {
    func longestPalindrome(_ s: String) -> Int {
        let frequency = s.frequency
        print(frequency)
        let evens = frequency.values.filter { $0 % 2 == 0 }.reduce(0, +)
        let odds = frequency.values.filter { $0 % 2 == 1 }.map { $0 - 1 }
        
        return odds.isEmpty ? evens : evens + odds.reduce(0, +) + 1
    }
}

extension Sequence where Element: Hashable {
    var frequency: [Element: Int] {
        let occurences = self.map { ($0, 1) }
        return Dictionary(occurences, uniquingKeysWith: +)
    }
}